src/fsharp/TypeChecker.fs

```diff
@@ -3043,6 +3043,27 @@ let BuildPossiblyConditionalMethodCall cenv env isMutable m isProp minfo valUseF
 let TryFindIntrinsicOrExtensionMethInfo (cenv: cenv) (env: TcEnv) m ad nm ty = 
     AllMethInfosOfTypeInScope cenv.infoReader env.NameEnv (Some(nm), ad) IgnoreOverrides m ty
 
+let TryFindFSharpSignatureInstanceGetterProperty (cenv: cenv) (env: TcEnv) m nm ty (sigTys: TType list) =
+    match TryFindPropInfo cenv.infoReader m env.AccessRights nm ty with
```

Comment:
I think you can refactor a bit and remove the outer pattern match. 

    TryFindPropInfo cenv.infoReader m env.AccessRights nm ty
    |> List.tryFind ...

Reply:
Yup, I definitely can